newIDE/app/src/InstancesEditor/InstancesRenderer/LayerRenderer.js

```diff
@@ -134,22 +143,93 @@ export default class LayerRenderer {
     );
   };
 
-  getInstanceWidth = (instance: gdInitialInstance) => {
+  getUnrotatedInstanceWidth = (instance: gdInitialInstance) => {
     if (instance.hasCustomSize()) return instance.getCustomWidth();
 
     return this.renderedInstances[instance.ptr]
       ? this.renderedInstances[instance.ptr].getDefaultWidth()
       : 0;
   };
 
-  getInstanceHeight = (instance: gdInitialInstance) => {
+  getUnrotatedInstanceHeight = (instance: gdInitialInstance) => {
     if (instance.hasCustomSize()) return instance.getCustomHeight();
 
     return this.renderedInstances[instance.ptr]
       ? this.renderedInstances[instance.ptr].getDefaultHeight()
       : 0;
   };
 
+  getUnrotatedInstanceAABB(
+    instance: gdInitialInstance,
+    bounds: Rectangle
+  ): Rectangle {
+    const left = this.getUnrotatedInstanceLeft(instance);
+    const top = this.getUnrotatedInstanceTop(instance);
+    const right = left + this.getUnrotatedInstanceWidth(instance);
+    const bottom = top + this.getUnrotatedInstanceHeight(instance);
+
+    bounds.left = left;
+    bounds.right = right;
+    bounds.top = top;
+    bounds.bottom = bottom;
+    return bounds;
+  }
+
+  _getRectanglePath(instance: gdInitialInstance): Array<[number, number]> {
```

Comment:
As this is used only in _getRotatedRectangle, I would exceptionally put this back inside `_getRotatedRectangle `